x-pack/test/observability_ai_assistant_api_integration/tests/public_complete/public_complete.spec.ts

```diff
@@ -0,0 +1,232 @@
+          .post(PUBLIC_COMPLETE_API_URL)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            messages,
+            connectorId,
+            persist: true,
+            actions: params.actions,
+            instructions: params.instructions,
+          })
+          .end((err, response) => {
+            if (err) {
+              return reject(err);
+            }
+            return resolve(response);
+          });
+      });
+
+      const [conversationSimulator, titleSimulator] = await Promise.all([
+        conversationInterceptor.waitForIntercept(),
+        titleInterceptor.waitForIntercept(),
+      ]);
+
+      await titleSimulator.status(200);
+      await titleSimulator.next('My generated title');
+      await titleSimulator.complete();
+
+      await conversationSimulator.status(200);
+      await cb(conversationSimulator);
+
+      const response = await responsePromise;
+
+      return String(response.body)
+        .split('\n')
+        .map((line) => line.trim())
+        .filter(Boolean)
+        .map((line) => JSON.parse(line) as StreamingChatResponseEvent)
+        .slice(2); // ignore context request/response, we're testing this elsewhere
+    }
+
+    before(async () => {
+      proxy = await createLlmProxy(log);
+
+      const response = await supertest
+        .post('/api/actions/connector')
+        .set('kbn-xsrf', 'foo')
+        .send({
+          name: 'OpenAI Proxy',
+          connector_type_id: '.gen-ai',
+          config: {
+            apiProvider: 'OpenAI',
+            apiUrl: `http://localhost:${proxy.getPort()}`,
+          },
+          secrets: {
+            apiKey: 'my-api-key',
+          },
+        })
+        .expect(200);
+
+      connectorId = response.body.id;
+    });
+
+    after(async () => {
+      await supertest
+        .delete(`/api/actions/connector/${connectorId}`)
+        .set('kbn-xsrf', 'foo')
+        .expect(204);
+
+      proxy.close();
+    });
+
+    describe('after executing an action', async () => {
+      let events: StreamingChatResponseEvent[];
+
+      before(async () => {
+        events = await getEvents(
+          {
+            actions: [
+              {
+                name: 'my_action',
+                description: 'My action',
+                parameters: {
+                  type: 'object',
+                  properties: {
+                    foo: {
+                      type: 'string',
+                    },
+                  },
+                },
+              },
+            ],
+          },
+          async (conversationSimulator) => {
+            await conversationSimulator.next({
+              function_call: { name: 'my_action', arguments: JSON.stringify({ foo: 'bar' }) },
+            });
+            await conversationSimulator.complete();
+          }
+        );
+      });
+
+      it('closes the stream without persisting the conversation', () => {
+        expect(
+          pick(
+            events[events.length - 1],
```

Comment:
Nit: `_.last(events)`?

Reply:
mmmeeh, I then have to use `!` because `last()` (correctly!) returns a nullable, but I know in this case the event is there. I'll keep this one as-is!